packages/contracts-bedrock/scripts/Deploy.s.sol

```diff
@@ -103,7 +103,7 @@ contract Deploy is Deployer {
 
     /// @notice Modifier that wraps a function in broadcasting.
     modifier broadcast() {
-        vm.startBroadcast();
+        vm.startBroadcast(msg.sender);
```

Comment:
> Pass `msg.sender` to `vm.broadcast`. This makes the `Deploy` contract usable in another script.

Can you expand on why this is needed? It seems like a no-op, since I think:
- If you invoke `forge script --private-key <key>` both formats (`vm.startBroadcast()` and `vm.startBroadcast(msg.sender)`) will result in a broadcast from the account with the given private key
- If you invoke `forge test`, both formats will result in the broadcast acting like a prank from the default forge sender

Reply:
This is required for when importing `Deploy` into another contract and calling `run` and having the correct account be creating the transactions. Without this, it was making the foundry default sender do the broadcasting

Reply:
Without this change, calling `run()` from another contract reverts:

```
    ├─ [362405] Deploy::run() 
    │   ├─ [0] console::log(Deploying L1 system) [staticcall]
    │   │   └─ ← ()
    │   ├─ [0] VM::startBroadcast() 
    │   │   └─ ← ()
    │   ├─ [327884] → new AddressManager@0x90193C961A926261B756D1E5bb255e67ff9498A1
    │   │   ├─ emit OwnershipTransferred(previousOwner: 0x0000000000000000000000000000000000000000, newOwner: DefaultSender: [0x1804c8AB1F12E6bbf3894d4083f33e07309d1f38])
    │   │   └─ ← 1519 bytes of code
    │   ├─ [307] AddressManager::owner() [staticcall]
    │   │   └─ ← DefaultSender: [0x1804c8AB1F12E6bbf3894d4083f33e07309d1f38]
    │   └─ ← "EvmError: Revert"
```

It fails on the following line:
https://github.com/ethereum-optimism/optimism/blob/93283d25adbe2e01e003892bd3886053a2052c73/packages/contracts-bedrock/scripts/Deploy.s.sol#L202

Reply:
Sorry I might be a bit rusty on broadcast behavior—how does this change result in the `manager.owner() == msg.sender` check passing during tests though, wouldn't you need to run with `forge test --sender <manager.owner>`?

Reply:
> Without this change, calling `run()` from another contract reverts:

Oh I think I see now: the assumption here is we're moving to an architecture where the test contract deploys an instance of the script contract and makes a call to `run()`, so we need to prank so that call has the correct sender address?

That's interesting, and does makes sense. I was thinking we'd do it via inheritance, where the test inherits from the deploy script, like this: https://github.com/ScopeLift/foundry-template/blob/fd3875d2e99a65dec19431723d6516b4ed76746e/test/Counter.t.sol#L8-L12

Reply:
> I was thinking we'd do it via inheritance, where the test inherits from the deploy script

 Not strongly opinionated  either way, but FWIW I was under the same impression.